src/ai/backend/test/testcases/root.py

```diff
@@ -0,0 +1,44 @@
+import uuid
+from contextlib import asynccontextmanager as actxmgr
+from contextvars import ContextVar
+from typing import Any, AsyncIterator, Optional
+
+from ai.backend.test.testcases.context import BaseTestContext
+from ai.backend.test.testcases.template import TestTemplate, WrapperTestTemplate
+
+
+class TestRunContext(BaseTestContext[dict[str, Any]]):
+    """Context for the entire test run"""
+
+    _ctxvar: ContextVar[Optional[dict[str, Any]]] = ContextVar("test_run_context", default=None)
+
+    @classmethod
+    def get_current(cls) -> Any:
+        return super().get_current()
+
+    @classmethod
+    def get_test_id(cls) -> str:
+        return cls.get_current()["test_id"]
```

Comment:
This part doesn't seem necessary.
And in this situation, **please don't specifically receive `dict` as a type.**
Just using contextvars already makes tracking difficult, but tracking dynamic fields with `dict` practically makes tracking impossible.